src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/ExtensibilityTelemetryCollector.cs

```diff
@@ -246,4 +245,14 @@ protected override void OnEventWritten(EventWrittenEventArgs eventData)
             }
         }
     }
+
+    internal static class DeconstructionExtensions
+    {
+        [DebuggerStepThrough]
+        public static void Deconstruct<TKey, TValue>(this KeyValuePair<TKey, TValue> pair, out TKey key, out TValue value)
```

Comment:
Why is this needed? 😕

Reply:
There's a few `foreach` loops in this class that deconstructs the KVP into a value tuple.